js/controller.js

```diff
@@ -635,6 +641,40 @@ OO.plugin("Html5Skin", function (OO, _, $, W) {
         this.state.mainVideoDuration = this.state.duration;
       }
     },
+  
+    onTouchMove: function(params) {
+      if (this.videoVr) {
+        this.mb.publish(OO.EVENTS.TOUCH_MOVE, this.focusedElement, params);
+      }
+    },
+  
+    checkVrDirection: function () {
+      if (this.videoVr) {
+        this.mb.publish(OO.EVENTS.CHECK_VR_DIRECTION, this.focusedElement);
+      }
+    },
+  
+    setViewingDirection: function(event, yaw, roll, pitch) {
+      this.state.viewingDirection = {yaw: yaw, roll: roll, pitch: pitch};
+    },
+  
+    recreatingUI: function (event, elementId, params, settings) {
+      if (!$('.oo-player-skin').length) {
+        this.state.mainVideoInnerWrapper.append("<div class='oo-player-skin'></div>")
+      }
+  
+      //load player with page level config param if exist
+      if (params.skin && params.skin.config) {
+        $.getJSON(params.skin.config, function(data) {
```

Comment:
Is it necessary to request skin.json again? It looks like `this.state.customSkinJSON`is treated as read-only an can probably be re-used. We should confirm whether this is the case.